samples/cpp/classification_sample_async/main.cpp

```diff
@@ -84,37 +84,62 @@ int main(int argc, char* argv[]) {
-        input_info.model().set_layout("NCHW");
-        // 4) output() with no args assumes a model has a single result
-        // - output() with no args assumes a model has a single result
-        // - precision of tensor is supposed to be 'f32'
-        ppp.output().tensor().set_element_type(ov::element::f32);
-
-        // 5) Once the build() method is called, the pre(post)processing steps
-        // for layout and precision conversions are inserted automatically
-        model = ppp.build();
+        bool isNetworkCompiled = fileExt(FLAGS_m) == "blob";
+
+        ov::CompiledModel compiledModel;
+        const ov::Layout tensor_layout{ "NHWC" }; 
+
+        if (!isNetworkCompiled)
+        {
+            std::shared_ptr<ov::Model> model = core.read_model(FLAGS_m);
+            printInputAndOutputsInfo(*model);
+
+            OPENVINO_ASSERT(model->inputs().size() == 1, "Sample supports models with 1 input only");
+            OPENVINO_ASSERT(model->outputs().size() == 1, "Sample supports models with 1 output only");
+
+            // -------- Step 3. Configure preprocessing --------
+            ov::preprocess::PrePostProcessor ppp(model);
+            // 1) input() with no args assumes a model has a single input
+            ov::preprocess::InputInfo& input_info = ppp.input();
+            // 2) Set input tensor information:
+            // - precision of tensor is supposed to be 'u8'
+            // - layout of data is 'NHWC'
+            input_info.tensor().set_element_type(ov::element::u8).set_layout(tensor_layout);
+            // 3) Here we suppose model has 'NCHW' layout for input
+            input_info.model().set_layout("NCHW");
+            // 4) output() with no args assumes a model has a single result
+            // - output() with no args assumes a model has a single result
+            // - precision of tensor is supposed to be 'f32'
+            ppp.output().tensor().set_element_type(ov::element::f32);
+
+            // 5) Once the build() method is called, the pre(post)processing steps
+            // for layout and precision conversions are inserted automatically
+            model = ppp.build();
+
+            // -------- Step 6. Loading model to the device --------
+            slog::info << "Loading model to the device " << FLAGS_d << slog::endl;
+            
+            compiledModel = core.compile_model(model, FLAGS_d);
+        }
+        else
+        {
+            std::ifstream modelStream(FLAGS_m, std::ios_base::binary | std::ios_base::in);
+            if (!modelStream.is_open()) {
+                throw std::runtime_error("Cannot open model file " + FLAGS_m);
+            }
+            compiledModel = core.import_model(modelStream, FLAGS_d, {});
+            modelStream.close();
 
+            // The following is needed to achieve accuracy. 
+            if (FLAGS_d == "VPUX")
+                compiledModel.input().get_tensor().set_partial_shape({ 1,224,224,3 });
```

Comment:
If this doesn't apply to all the classification network models, can we address it through compile_tool config?